### ISSUE TITLE
Add Backstage plugin

### ISSUE THREAD

**Issue by Oshratn** [kubescape/kubescape#1667]
## Overview
A proof of concept plugin was built in the past, but never developed beyond PoC.

## Problem
 With the advent of the new Backstage plugin system the plugin requires a rewrite.

## Solution
A new Kubescape plugin for backstage that adheres to the new plugin system and reflects Kubescape's new capablities.

**Comment by ShivamJha2436** [comment]
Hello @Oshratn, 
I'm Shivam, a developer keen to contribute through LFX. The Kubescape Backstage plugin project excites me! My experience with TypeScript and React aligns well with the plugin's development. While I'm new to the Backstage plugin system, I'm eager to learn and contribute to this valuable tool. Participating in this project would be a fantastic learning opportunity to grow alongside experienced developers and make Kubescape even more accessible.
Thanks,
Shivam

**Comment by jayesh9747** [comment]
@Oshratn can you explain more about this project ?

**Comment by kunalworldwide** [comment]
we'd actually use this pretty heavily — devs keep asking why security findings live in a separate tool instead of being visible where they're already working. the new plugin system from the article, is that the new frontend plugin model from Backstage 1.20+ or still on the older pattern? tried adapting the old PoC plugin to the new system last quarter and the migration was messier than expected, mainly around the entity provider setup.

**Comment by matthyx** [comment]
@kunalworldwide I think we can consider it being the new plugin model from 1.20+
We don't have manpower to work on that and we prefer having one good UI (the headlamp plugin) than 2 so-so.
But if you want to work on that, I can review your work.

**Comment by kunalworldwide** [comment]
@matthyx makes sense to focus on one good UI rather than spreading thin. The new Backstage frontend plugin model (1.20+) would be the way to go if anyone picks this up — the older extension points are being deprecated. Noted on the headlamp plugin priority.

**Comment by mohityadav8** [comment]
@matthyx Started on this - scaffolded an initial Backstage plugin using the new frontend plugin system (1.20+): https://github.com/mohityadav8/kubescape-backstage-plugin

Went with the operator-CRD approach rather than the httphandler microservice, to mirror the headlamp-plugin's data model (workloadconfigurationscans, vulnerabilitymanifests) - figured that's the more relevant one given it matches Kubescape's current continuous-scan capabilities.

This is just the scaffold with a compliance summary card and a vulnerabilities table stubbed in. Wanted to check the direction is right before building it out further ..let me know if this approach makes sense or if you'd prefer something closer to the old PoC's design.

**Comment by matthyx** [comment]
well if you have it already, let's use the CRD approach you suggest

**Comment by mohityadav8** [comment]
@matthyx wanted to check - once this is full done(with testing all other factor i am still working on this ), what's the right way to submit it for review? Should I open a PR against this repo somehow, or would you prefer transferring/linking the plugin repo under the kubescape org, or something else?

**Comment by matthyx** [comment]
here https://github.com/kubescape/backstage-plugin

**Comment by mohityadav8** [comment]
@matthyx close this is this completed via https://github.com/kubescape/backstage-plugin/pull/4
